JavaScript/FastNoiseLite.js

```diff
@@ -0,0 +1,3372 @@
+		this.#CalculateFractalBounding();
+	}
+
+	/**
+	 * @description Sets octave weighting for all none DomainWarp fratal types
+	 * @remarks Default: 0.0 | Keep between 0...1 to maintain -1...1 output bounding
+	 * @default 0.5
+	 * @param {number} weightedStrength
+	 */
+	SetFractalWeightedStrength(weightedStrength) {
+		this.#WeightedStrength = weightedStrength;
+	}
+
+	/**
+	 * @description Sets strength of the fractal ping pong effect
+	 * @remarks Default: 2.0
+	 * @default 2.0
+	 * @param {number} pingPongStrength
+	 */
+	SetFractalPingPongStrength(pingPongStrength) {
+		this.#PingPongStrength = pingPongStrength;
+	}
+
+	/**
+	 * @description Sets distance function used in cellular noise calculations
+	 * @remarks Default: EuclideanSq 
+	 * @default FastNoise.CellularDistanceFunction.EuclideanSq
+	 * @param {FastNoise.CellularDistanceFunction} cellularDistanceFunction
+	 */
+	SetCellularDistanceFunction(cellularDistanceFunction) {
+		this.#CellularDistanceFunction = cellularDistanceFunction;
+	}
+
+	/**
+	 * @description Sets return type from cellular noise calculations
+	 * @remarks Default: Distance
+	 * @default FastNoise.CellularReturnType.Distance
+	 * @param {FastNoise.CellularReturnType} cellularReturnType
+	 */
+	SetCellularReturnType(cellularReturnType) {
+		this.#CellularReturnType = cellularReturnType;
+	}
+
+	/**
+	 * @description Sets the maximum distance a cellular point can move from it's grid position
+	 * @remarks Default: 1.0
+	 * @default 1.0
+	 * @param {number} cellularJitter
+	 */
+	SetCellularJitter(cellularJitter) {
+		this.#CellularJitterModifier = cellularJitter;
+	}
+
+	/**
+	 * @description Sets the warp algorithm when using DomainWarp(...)
+	 * @remarks Default: OpenSimplex2
+	 * @default FastNoise.DomainWarpType.OpenSimplex2
+	 * @param {FastNoise.DomainWarpType} domainWarpType
+	 */
+	SetDomainWarpType(domainWarpType) {
+		this.#DomainWarpType = domainWarpType;
+		this.#UpdateWarpTransformType3D();
+	}
+
+	/**
+	 * @description Sets the maximum warp distance from original position when using DomainWarp(...)
+	 * @remarks Default: 1.0
+	 * @default 1.0
+	 * @param {number} domainWarpAmp
+	 */
+	SetDomainWarpAmp(domainWarpAmp) {
+		this.#DomainWarpAmp = domainWarpAmp;
+	}
+
+	/**
+	 * @description 2D/3D noise at given position using current settings
+	 * @param {number} x X coordinate
+	 * @param {number} y Y coordinate
+	 * @param {number} [z] Z coordinate
+	 * @return {number} Noise output bounded between -1...1
+	 */
+	GetNoise(x, y, z) {
+		/**
+		 * @description 2D noise at given position using current settings
+		 * @param {number} x
+		 * @param {*} y
+		 * @return {number} Noise output bounded between -1...1
+		 */
+		let R2 = (x, y) => {
+			x *= this.#Frequency;
+			y *= this.#Frequency;
+
+			switch (this.#NoiseType) {
+				case FastNoise.NoiseType.OpenSimplex2:
+				case FastNoise.NoiseType.OpenSimplex2S:
+					const SQRT3 = 1.7320508075688772935274463415059;
+					const F2 = 0.5 * (SQRT3 - 1);
```

Comment:
Do you know if these method-level constant declarations are re-computed on every call, or if they're pre-evaluated during method declaration parsing? I wonder if it would make a speed difference to move them to class level. I'm having trouble finding documentation on this. If you don't know, I probably wouldn't worry about it.

Reply:
I dont know...

Reply:
Another thing to keep in mind: `switch` case blocks do not open a new lexical scope. All cases with declarations (`const`, `let`, ...) should be wrapped in blocks to force a new lexical scope.

See: https://eslint.org/docs/rules/no-case-declarations